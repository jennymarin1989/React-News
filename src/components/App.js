import React, { Component } from "react";

class App extends Component{

  constructor(){
    super();
    this.state = {
      input: ''
    }
  };

  render() {
    return(
      <div className="news">
        <h1> A B C  News </h1>
        <div className="input">
          <input type="text" id="input-field" className="form-control" name="input" placeholder="What news are you looking for.."  />
        </div>
      </div>
    )
  }
}

export default App;