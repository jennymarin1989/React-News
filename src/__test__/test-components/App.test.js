import React from "react";
import { shallow } from "enzyme";
import renderer from 'react-test-renderer';
import App from '../../components/App';


describe('App', ()=> {

  describe('renders components correctly', ()=> {
    it('renders correctly', () => {
      const app = shallow(<App />);
      expect(app).toMatchSnapshot();
    });
  });

  describe('initial state ', () =>{
    it('initializes state successfully', () =>{
      const app = shallow(<App />);
      expect(app.state('input')).toEqual("");
    });
  });  

  describe('input text field', () => {
    it('finds input text field', () => {
      const app = shallow(<App />);
      expect(app.find('#input-field').length).toEqual(1);
    })
  })

})