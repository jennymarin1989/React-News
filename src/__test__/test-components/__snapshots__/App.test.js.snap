// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders components correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           A B C  News 
</h1>,
        <div
          className="input"
>
          <input
                    className="form-control"
                    id="input-field"
                    name="input"
                    placeholder="What news are you looking for.."
                    type="text"
          />
</div>,
      ],
      "className": "news",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " A B C  News ",
        },
        "ref": null,
        "rendered": " A B C  News ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="form-control"
            id="input-field"
            name="input"
            placeholder="What news are you looking for.."
            type="text"
/>,
          "className": "input",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "form-control",
            "id": "input-field",
            "name": "input",
            "placeholder": "What news are you looking for..",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             A B C  News 
</h1>,
          <div
            className="input"
>
            <input
                        className="form-control"
                        id="input-field"
                        name="input"
                        placeholder="What news are you looking for.."
                        type="text"
            />
</div>,
        ],
        "className": "news",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " A B C  News ",
          },
          "ref": null,
          "rendered": " A B C  News ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="input-field"
              name="input"
              placeholder="What news are you looking for.."
              type="text"
/>,
            "className": "input",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "input-field",
              "name": "input",
              "placeholder": "What news are you looking for..",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
